src/utils/activity/ExecuteActivityOutputStepBase.ts

```diff
@@ -0,0 +1,47 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtTreeItem, AzureWizardExecuteStep, ExecuteActivityContext, IActionContext } from "@microsoft/vscode-azext-utils";
+import { Progress } from "vscode";
+import { tryCatchActivityWrapper } from "../../utils/activity/activityUtils";
+
+export interface ExecuteActivityOutput {
+    /**
+     * The activity child item to display on success or fail
+     */
+    item?: AzExtTreeItem;
+    /**
+     * The output log message(s) to display on success or fail
+     */
+    output?: string | string[];
+}
+
+export interface ExecuteActivityOutputOptions {
+    shouldSwallowError?: boolean;
+}
+
+/**
+ * An execute activity base step (wrapper) that automatically handles displaying activity children and/or output log messages on success or fail
+ */
+export abstract class ExecuteActivityOutputStepBase<T extends IActionContext & ExecuteActivityContext> extends AzureWizardExecuteStep<T> {
+    abstract priority: number;
+    protected options: ExecuteActivityOutputOptions = {};
+
+    protected success: ExecuteActivityOutput;
+    protected fail: ExecuteActivityOutput;
+
+    public async execute(context: T, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void> {
+        this.success = this.initSuccessOutput(context);
+        this.fail = this.initFailOutput(context);
+
+        await tryCatchActivityWrapper(() => this.executeCore(context, progress), context, this.success, this.fail, this.options);
+    }
+
+    protected abstract executeCore(context: T, progress: Progress<{ message?: string | undefined; increment?: number | undefined }>): Promise<void>;
+    abstract shouldExecute(context: T): boolean;
+
+    protected abstract initSuccessOutput(context: T): ExecuteActivityOutput;
```

Comment:
We talked about refactoring it, but I honestly don't have a very strong opinion about it. I think this does make it a bit more clear about the success and fail states. I think with my comment above, I'd just do the try/catch in the `execute` like this:

```
try {
  await this.executeCore(context, progress), context, this.options);
  output this.initSuccessOutput(context)
} catch (theseHands) {
  output this.initFailOutput(context)
}
```

I think after seeing this base step, that I agree that this is more clear. Though I don't really like the naming-- not sure what it is "initializing" really. I think createFailOutput or createFailChild maybe makes more sense to me?